tcav/imagenet_and_broden_fetcher.py

```diff
@@ -0,0 +1,268 @@
+import pandas as pd
+import urllib.request
+import os
+import shutil
+import PIL
+from PIL import Image
+import tensorflow as tf
+import socket
+import random
+
+kImagenetBaseUrl = "http://imagenet.stanford.edu/api/text/imagenet.synset.geturls?wnid="
+kBrodenTexturesPath = "broden1_224/images/dtd/"
+kMinFileSize = 10000
+
+####### Helper functions
+""" Makes a dataframe matching imagenet labels with their respective url.
+
+Reads a csv file containing matches between imagenet synids and the url in
+which we can fetch them. Appending the synid to kImagenetBaseUrl will fetch all
+the URLs of images for a given imagenet label
+
+  Args: path_to_imagenet_classes: String. Points to a csv file matching
+          imagenet labels with synids.
+
+  Returns: a pandas dataframe with keys {url: _ , class_name: _ , synid: _}
+"""
+def make_imagenet_dataframe(path_to_imagenet_classes):
+  urls_dataframe = pd.read_csv(path_to_imagenet_classes)
+  urls_dataframe["url"] = kImagenetBaseUrl + urls_dataframe["synid"]
+  return urls_dataframe
+
+""" Downloads an image.
+
+Downloads and image from a image url provided and saves it under path.
+Filters away images that are corrupted or smaller than 10KB
+
+  Args:
+    path: Path to the folder where we're saving this image.
+    url: url to this image.
+
+  Raises:
+    Exception: Propagated from PIL.image.verify()
+"""
+
```

Comment:
remove empty lines between function doc and def line, in all functions below.

Reply:
Done